app/src/main/java/com/example/meetup/SignUpActivity.java

```diff
@@ -0,0 +1,92 @@
+package com.example.meetup;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AlertDialog;
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.app.ProgressDialog;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.Toast;
+
+import com.google.android.gms.auth.api.signin.GoogleSignIn;
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
+import com.google.android.material.textfield.TextInputEditText;
+import com.parse.ParseException;
+import com.parse.ParseUser;
+
+import com.parse.SignUpCallback;
+
+public class SignUpActivity extends AppCompatActivity {
+
+    private ImageView back;
+    private Button signUp;
+    private TextInputEditText username;
+    private TextInputEditText password;
+    private TextInputEditText passwordAgain;
+    private ProgressDialog progressDialog;
+    private GoogleSignInAccount account;
+
```

Comment:
Interesting. The android studio should auto format the file so there is no extra empty line. Usually there should be only one empty line between the code blocks.

Reply:
Let's explore to see if android studio can auto format this when you save the file.